tests/unit/include/bout/test_macro_for_each.cxx

```diff
@@ -0,0 +1,86 @@
+#include "gtest/gtest.h"
+
+#include "bout/macro_for_each.hxx"
+
+#include <sstream>
+
+#define INC(x) {++(x);}
+
+// Test that the macro is expanded
+// the correct number of times
+TEST(MacroForEachTest, ExpandCount) {
+  int a{0};
+
+  MACRO_FOR_EACH(INC, a);
+  EXPECT_EQ(a, 1);
+
+  a = 0;
+  MACRO_FOR_EACH(INC, a, a);
+  EXPECT_EQ(a, 2);
+
+  // At least 10 arguments supported
+  a = 0;
+  MACRO_FOR_EACH(INC, a, a, a, a, a, a, a, a, a, a);
+  EXPECT_EQ(a, 10);
+}
+
+#define PUSH_NAME(var) ss << #var;
+
+// Check that the macro applies in the order given
+TEST(MacroForEachTest, Order) {
+  std::stringstream ss;
+
+  MACRO_FOR_EACH(PUSH_NAME, a, b, c);
+  EXPECT_EQ(ss.str(), "abc");
+}
+
+// No braces are put around the expansion
+// This is needed in some cases, but can
+// lead to unexpected behaviour.
+TEST(MacroForEachTest, NoBraces) {
+  int a = 0;
+
+  // Only the first expansion is disabled
+  if(false)
```

Comment:
Nice!

Reply:
Maybe a stupid question, my brain is not fully in gear -- if this conditional gets optimised away, that's still proof that the macro is working, right?